drivers/platform/generic/timer.c

```diff
@@ -0,0 +1,184 @@
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include "error.h"
+#include "timer.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @breif Initialize hardware timer and the handler structure associated with
+ *        it.
+ * @param [out] desc - Pointer to the reference of the device handler.
+ * @param [in] init_param - Initialization structure.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_init(struct timer_desc **desc,
+		   struct timer_init_param *init_param)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+	if(init_param) {
+		// Unused variable - fix compiler warning
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Free the memory allocated by timer_setup().
+ * @param [in] desc - Pointer to the device handler.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_remove(struct timer_desc *desc)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Start a timer.
+ * @param [in] desc - Pointer to the device handler.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_start(struct timer_desc *desc)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+}
+
+/**
+ * @brief Stop a timer from counting.
+ * @param [in] desc - Pointer to the device handler.
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t timer_stop(struct timer_desc *desc)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+}
+
+/**
+ * @brief Get the value of the counter register for the timer.
+ * @param [in]  desc    - Pointer to the device handler.
+ * @param [out] counter - Pointer to the counter value.
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t timer_counter_get(struct timer_desc *desc, uint32_t *counter)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+
+	if(counter) {
+		// Unused variable - fix compiler warning
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Set the timer counter register value.
+ * @param [in] desc    - Pointer to the device handler.
+ * @param [in] new_val - The new value of the counter register.
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t timer_counter_set(struct timer_desc *desc, uint32_t new_val)
+{
+	if(desc) {
+		// Unused variable - fix compiler warning
+	}
+
+	if(new_val) {
```

Comment:
I know this is just a template implementation, but this comparison is always true ;)

Reply:
This will not be true if `new_val` is 0.

Reply:
For some reason I wanted to comment about this check being pointless (looks like a copy-paste from the other functions where the second parameter is a pointer and is checked against NULL) and I commented something completely wrong hah :D 

What I mean is, with this function, any value should possible, the user can set the counter to any value from uint32_t including 0 so the test is pointless.

Reply:
Well, this is not actually a test - it is only a workaround for fixing the compiler warning.

Reply:
Oh ok, I totally misunderstood the point although there is a comment "fix compiler warning".